server/model/monitor.js

```diff
@@ -1461,6 +1496,110 @@ class Monitor extends BeanModel {
         }
     }
 
+    /**
+     * Check expiry date based on RDAP
+     * @param {string} domain Domain to lookup
+     * @returns {void}
+     */
+    async checkDomainExpiryNotifications(domain) {
+        const notificationList = await Monitor.getNotificationList(this);
+        if (! notificationList.length > 0) {
+            // fail fast. If no notification is set, all the following checks can be skipped.
+            log.debug("monitor", "No notification, no need to send domain notification");
+            return;
+        }
+
+        let domainInfoBean = await R.findOne("domain_expiry_info", "domain = ?", [ domain ]);
+
+        let expiryDate;
+        if (domainInfoBean === null || getDaysBetween(new Date(domainInfoBean.lastCheck), new Date()) > 1) {
```

Comment:
Let's move this logic to the top via inverting the check and doing an early return

I think this check is more selective.

Reply:
I meant pushing `notificationList` intialisation below `expiryDate === null`.

I the end likely does not matter all that much.
Feel free to ignore 😉

Reply:
Oh ok I see now, I can swap the two 👍

Reply:
I will let the order like this, otherwise the conditions would be too messy